src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,46 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private Node<T> element;
```

Comment:
element is to abstract, this field is not just random element it's last/top or smt like this, it should clear that this element is not from the middle or random place